hw4/04-templates.js

```diff
@@ -5,60 +5,96 @@ const app = express();
 const port = process.env.PORT || 5000;
 
 // Use Pug as the templating engine
-app.set('views', __dirname + '/views');
+app.set('views', `${__dirname}/views`);
 app.set('view engine', 'pug');
 
 // REST Countries URL
 const url = 'https://restcountries.com/v3.1/all';
 
 // Add your code here
-
-app.get('/', (req, res) => {
+const queryCountries = async (queryUrl) => axios.get(queryUrl, { }).then();
+const localizedComparator = (a, b) => a.localeCompare(b);
+const populationComparator = (a, b) => {
+  if (a.population > b.population) {
+    return -1;
+  }
+  if (a.population < b.population) {
+    return 1;
+  }
+  return 0;
+};
+
+app.get('/', async (req, res) => {
   // render pug template for the index.html file
-
   res.render('index', {
     heading: 'Countries of the World',
     main: 'Welcome to this application. Using the REST Countries API, we will be showing the countries and capitals of the world, the most populous countries in the world, and the number of countries in each region of the world',
   });
 });
-
-app.get('/capitals', (req, res) => {
+app.get('/capitals', async (req, res) => {
   // map the output array to create an array with country names and capitals
   // check for empty data in the output array
-
-  let countries = ['Afghanistan', 'Aland Islands', 'Albania'];
-
+  const countries = (await queryCountries(url)).data;
+  for (let i = 0; i < countries.length; i += 1) {
+    const country = countries[i];
+    const countryName = country.name.common;
+    let capital = '';
+    if (country.capital) [capital] = country.capital;
```

Comment:
You could combine 
1) initialization
2) conditional assignment
here with the optional chaining operator:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

and the logical or operator


```javascript 
const capital = country?.capital || ''
```